torch/onnx/__init__.py

```diff
@@ -388,8 +388,18 @@ def forward(self, x):
             fallback=fallback,
         )
     else:
+        import warnings
+
         from torch.onnx.utils import export
 
+        warnings.warn(
+            "The legacy ONNX export logic is deprecated and will be removed in a future release. "
+            "Please use the new export logic by setting `dynamo=True` in the `torch.onnx.export` API "
+            "for better performance and support for modern features.",
```

Comment:
I wonder if we should provide more detail of the difference between dynamo=True/False in this message. For example, a heads up for them to use dynamic_shapes, instead of dynamic_axes, and they need to use torch.cond/torch.scan.

Reply:
Maybe link them to a docs page?

Reply:
Sounds right. I suggest we add one for it